### PR TITLE
remove workspace

### LINKED ISSUE #10351: make `workspace()` reload `.juliarc.jl`
In the manual/FAQ/discussions, `workspace()` is the recommended way of getting a "clean slate", that should be equivalent to restarting Julia (except for `LastMain` etc).

Arguably, `workspace()` then should also reload `.juliarc.jl` in certain cases.

For example, having an optional `reload-startup` argument would make sense, with the following options:
1. `:auto`, which is the default. Reload unless the process was started with `--startup-file=no`, or `-f`.
2. `false`: never reload.
3. `true`: reload unconditionally.

### LINKED ISSUE #22434: constant redefinition after workspace() fails on master under some circumstances
Consider:

```julia
const X = 1
workspace()
const X = []
```

On master, putting this in a file and running it results in an error (`ERROR: LoadError: invalid redefinition of constant X`), although pasting it into the REPL works. It works in both cases on 0.6.

This functionality was useful for package tests, but the docs do say that "This function should only be used interactively." However, I can also make it fail at the REPL if that is desired:

```julia
julia> const X = 1; workspace(); const X = []
ERROR: invalid redefinition of constant X
```

### PR DESCRIPTION
**Pull request by fredrikekre**
close #10351, close #22434

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -732,6 +732,8 @@ Deprecated or removed
   * `trues(A::AbstractArray)` and `falses(A::AbstractArray)` are deprecated in favor of
     `trues(size(A))` and `falses(size(A))` respectively ([#24595]).
 
+  * `workspace` is discontinued ([#25046]).
```
Comment (StefanKarpinski) [inline NEWS.md]: Maybe a link to Revise?

**Review by StefanKarpinski — APPROVED** [review]

File: NEWS.md [hunk 2]
```diff
@@ -732,6 +732,9 @@ Deprecated or removed
   * `trues(A::AbstractArray)` and `falses(A::AbstractArray)` are deprecated in favor of
     `trues(size(A))` and `falses(size(A))` respectively ([#24595]).
 
+  * `workspace` is discontinued, check out [Revise.jl](https://github.com/timholy/Revise.jl)
+    for an alternative workflow ([#25046]).
```
Comment (iblis17) [inline NEWS.md]: 😱  Revise.jl still doesn't work on FreeBSD

Comment (iblis17) [reply NEWS.md]: issue filed.
https://github.com/timholy/Revise.jl/issues/66

Comment (Ismael-VC) [reply NEWS.md]: @StefanKarpinski shouldn't this needed a deprecation warning too, not just deleting it?

Else people will just get an `UndefVarError`, with no idea that they should be checking out Revise.

**Pull request merged**